### PR TITLE
feat: add switch component

### PR DESCRIPTION
**Pull request by MaximeJineMaraval**
## Changes description 🧑‍💻

Add switch component

## Context 🤔

https://github.com/Decathlon/vitamin-android/issues/11

## Checklist ✅

 - [x] I have reviewed the submitted code
 - [x] I have tested on a phone device/emulator
 - [x] I have tested on a tablet device/emulator
 - [x] I have tested on a kiosk device/emulator

#### Phone

Screenshot_1621609469
Screenshot_1621609476

#### Tablet

Screenshot_1621609431
Screenshot_1621609435

#### Kiosk
Screenshot_1621609277
Screenshot_1621609280
Screenshot_1621609349
Screenshot_1621609355

### CODE REVIEW

File: switches/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+If you don't want to inherit the full Vitamin theme (and all the Vitamin components) on your entire app, you can use this component in standalone.
+
+### Version 1
+
+Inherit the Base Vitamin theme in your app to have the right colors and fonts and link the Vitamin switch style.
+You can now use the switch component as seen previously.
+
+```kotlin
+implementation 'com.decathlon.android.foundation:<version>'
+implementation 'com.decathlon.android:switches:<version>'
```
Comment (GerardPaligot) [inline switches/README.md:45]: `com.decathlon.vitamin:switches-vitamin:<version>`

File: switches/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+If you don't want to inherit the full Vitamin theme (and all the Vitamin components) on your entire app, you can use this component in standalone.
+
+### Version 1
+
+Inherit the Base Vitamin theme in your app to have the right colors and fonts and link the Vitamin switch style.
+You can now use the switch component as seen previously.
+
+```kotlin
+implementation 'com.decathlon.android.foundation:<version>'
```
Comment (GerardPaligot) [inline switches/README.md:44]: You don't need this dependency. It is tagged as `api` in all components.

File: switches/README.md (hunk 1 above)
Comment (GerardPaligot) [reply switches/README.md:45]: Change in all versions

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

File: vitamin/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -20,5 +20,7 @@
         <!--  RadioButton -->
         <item name="radioButtonStyle">@style/Widget.Vitamin.RadioButton</item>
         <item name="android:radioButtonStyle">@style/Widget.Vitamin.RadioButton</item>
+        <!-- Switches -->
+        <item name="switchStyle">@style/Widget.Vitamin.Switch</item>
```
[the remark this answers is not in the document]
Comment (MaximeJineMaraval) [reply vitamin/src/main/res/values/styles.xml:24]: Which case are we supposed to cover with this legacy android property? 
With only `switchStyle` it works on the older Android API we support today (API 17)

File: switches/src/main/res/values/styles-switches.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="Widget.Vitamin.Switch" parent="Widget.MaterialComponents.CompoundButton.Switch">
+        <item name="materialThemeOverlay">@style/ThemeOverlay.Vitamin.Switch</item>
+        <item name="android:textAppearance">@style/TextAppearance.Vitamin.Body2</item>
+    </style>
+
+    <style name="ThemeOverlay.Vitamin.Switch" parent="">
+        <item name="colorOnSurface">@color/vtmn_black</item>
```
[the remark this answers is not in the document]
Comment (MaximeJineMaraval) [reply switches/src/main/res/values/styles-switches.xml:10]: Request sent at Sabrina

File: switches/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+## Switch design specs
+You can find the design specs on [www.decathlon.design](https://www.decathlon.design)
+
+## Usage with Vitamin Theme
+
+By inheriting the Vitamin theme in your app, you just have to use the default SwitchMaterial component or the VitaminSwitch component. 
+
+```kotlin
+implementation 'com.decathlon.android.vitamin:<version>'
```
[the remark this answers is not in the document]
Comment (MaximeJineMaraval) [reply switches/README.md]: Fixed

File: switches/README.md (hunk 1 above)
Comment (MaximeJineMaraval) [reply switches/README.md]: Fixed

File: switches/README.md (hunk 2 above)
Comment (MaximeJineMaraval) [reply switches/README.md]: Fixed

File: switches/README.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:checked="true"
+    android:text="My switch component" />
+```
+
+or 
+
+```xml
+<com.decathlon.android.switches.VitaminSwitch
```
Comment (GerardPaligot) [inline switches/README.md:26]: Move before the `SwitchMaterial` example

File: switches/README.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+<com.google.android.material.switchmaterial.SwitchMaterial
+    android:id="@+id/switch"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:checked="true"
+    android:text="My switch component" />
+```
+
+```xml
+<com.decathlon.android.switches.VitaminSwitch
```
Comment (GerardPaligot) [inline switches/README.md:66]: Move before `SwitchMaterial`

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

**Review by GerardPaligot — APPROVED** [review]

**Pull request merged**